sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestRecord.java

```diff
@@ -213,13 +214,26 @@ public long transportRequestId() {
     // region Methods
 
     public boolean expire() {
-        final RequestTimeoutException error = new RequestTimeoutException(this.toString(), this.args.physicalAddress());
+        final GoneException error = new GoneException(this.toString(), null, this.args.physicalAddress());
         BridgeInternal.setRequestHeaders(error, this.args.serviceRequest().getHeaders());
+
         return this.completeExceptionally(error);
     }
 
     public abstract Timeout newTimeout(final TimerTask task);
 
+    /**
+     * Provides information whether the request could have been sent to the service
+     * @return fals if it is possible to guarantee that the request never arrived at the service - true otherwise
+     */
+    public boolean hasSendingRequestStarted() {
+        return this.sendingRequestHasStarted;
+    }
+
+    void setSendingRequestHasStarted() {
+        this.sendingRequestHasStarted = true;
```

Comment:
I wonder if we should incorporate that into the RntbdRequestRecord stages as SEND_STARTED stage, and this as an event to the request timeline.

e.g.:
QUEUED, PIPELINED, SEND_STARTED, SENT, RECEIVED, COMPLETED

Reply:
I thought about it - decided against it because I haven't found a place where I could interrupt really when starting to send on the wire. Right now I set the flag at the last point where I can guarantee that sending hasn't started yet - which is good enough for the exception handling. But when pushing it into a first-class STAGE i think we should do it really when starting to send data to the wire. So my preference would to leave this as is - are you ok with it or still disagree?

Reply:
I agree with you. Thanks.